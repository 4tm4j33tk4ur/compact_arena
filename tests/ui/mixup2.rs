extern crate compact_arena;

use compact_arena::mk_nano_arena;

fn main() {
    mk_nano_arena!(a);
    mk_nano_arena!(b);
    let i = a.add(1usize);
    let j = b.add(1usize);
    let _ = b[i];
}
