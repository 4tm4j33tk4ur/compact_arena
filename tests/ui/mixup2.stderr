error[E0597]: `tag` does not live long enough
  --> $DIR/mixup2.rs:7:5
   |
7  |     mk_nano_arena!(b);
   |     ^^^^^^^^^^^^^^^^^^ borrowed value does not live long enough
...
11 | }
   | -
   | |
   | `tag` dropped here while still borrowed
   | borrow might be used here, when `_guard` is dropped and runs the `Drop` code for type `main::Guard`
   |
   = note: values in a scope are dropped in the opposite order they are defined
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0597`.
